Main.java

```diff
@@ -3,58 +3,51 @@
-    static final String REMOVE_SPECIAL_CHAR = "[^a-zA-Z0-9 -]";
-    static final String SINGLE_SPACE = "";
-    static String searchResult;
+    static String textFilePath = null;
+    static String searchKeyword = null;
 
     public static void main(String[] args) {
 
-        String textFilePath;
-        String searchKeyword;
-        String errorMessage;
-        if (args.length != 2) {
-            errorMessage = "Provide valid filepath and search keyword";
-            System.out.println(errorMessage);
-            Database.databaseOperation(null, null, searchResult = "Failure", errorMessage, 0);
+        if((args.length != 2) || (args[0] == null || args[0] == " " || args[0] == "") || (args[1] == null || args[1] == " " || args[1] == "")) {
+            Constants.errorMessage1 = "Both file path and Search word must contain some value";
+            System.out.println(Constants.errorMessage1);
+            DatabaseLink databaseLink = new DatabaseLink(textFilePath, searchKeyword, Constants.FAILURE ,Constants.errorMessage1, 0);
+            databaseLink.start();
             return;
         }
-        textFilePath = args[0];
-        searchKeyword = args[1];
-        File file = new File(textFilePath);
-        System.out.println("Processing...");
-        try {
-            if (isValidFile(file, textFilePath, searchKeyword)) {
+            textFilePath = args[0];
+            searchKeyword = args[1];
+            File file = new File(textFilePath);
+            System.out.println("Processing...");
+            if(isValidFile(file, textFilePath, searchKeyword)) {
                 if (file.length() != 0) {
                     wordCount(textFilePath, searchKeyword);
                 } else {
-                    errorMessage = "File does not contain any data";
-                    System.out.println(errorMessage);
-                    Database.databaseOperation(textFilePath, searchKeyword, searchResult = "Failure", errorMessage, 0);
+                    Constants.errorMessage2 = "File does not contain any data";
+                    System.out.println(Constants.errorMessage2);
+                    DatabaseLink databaseLink = new DatabaseLink(textFilePath, searchKeyword, Constants.FAILURE,Constants.errorMessage2, 0);
+                    databaseLink.start();
                 }
             }
-        } catch (ClassNotFoundException e) {
-            e.printStackTrace();
         }
-    }
 
-    private static boolean isValidFile(File file, String textFilePath, String searchKeyword) throws ClassNotFoundException {
-        String errorMessage;
+        private static boolean isValidFile(File file, String textFilePath, String searchKeyword) {
         if (!file.exists()) {
-            errorMessage = "File does not exist";
-            System.out.println(errorMessage);
-            Database.databaseOperation(textFilePath, searchKeyword, searchResult = "Failure", errorMessage, 0);
+            Constants.errorMessage3 = "File does not exist";

```

Comment:
You have already defined these messages as constants, then why are you reassigning them?

Reply:
Mistake from my part. Will remove them immediately.